parsers/AUS-TAS-KI.py

```diff
@@ -0,0 +1,67 @@
+import asyncio
+import websockets
+import json
+import logging
+import arrow
+
+technologies_parsed = {}
+
+def fetch_api(logger):
+    async def fetch():
+        # TODO this is likely not a viable endpoint / way to fetch this
+        uri = "wss://data.ajenti.com.au/live/signalr/reconnect?transport=webSockets&messageId=d-C8F39B96-km%2C152%7Ckr%2C0%7Cks%2C2&clientProtocol=1.5&connectionToken=Yu0BMoSk5rEBlaiSH%2BaLhoYTA3oiXDBcS%2FvqcVQr7q%2FnKYKbTppUSNS1Mz%2Bx1wwgH4VHAgSv5SalVBzsYrWiG92AM7r5qBJogpCJYS%2BqEWECTRQS%2BUl62HQ5UjFC6zCh&connectionData=%5B%7B%22name%22%3A%22taghub%22%7D%5D&tid=3"
+        async with websockets.connect(uri) as websocket:
+            payload_str = await websocket.recv()
+            payload = json.loads(payload_str)
+            for message in payload['M']:
+                for serie in message['A']:
+                    logger.debug(f"serie : {json.dumps(serie)}")
+                    for technology in serie['technologies']:
+                        assert technology['unit'] == 'kW'
+                        # The upstream API gives us kW, we need MW
+                        technologies_parsed[technology['id']] = int(technology['value'])/1000
+                    logger.debug(f"production : {json.dumps(technologies_parsed)}")
+
+    asyncio.get_event_loop().run_until_complete(fetch())
+    return technologies_parsed
+
+def fetch_production(zone_key='AUS-TAS-KI', session=None, target_datetime=None, logger: logging.Logger = logging.getLogger(__name__)):
+
+    if target_datetime is not None:
+        raise NotImplementedError('The datasource currently implemented is only real time')
+  
+    if session is not None:
+        # TODO : is that the best to do in our case ? (websockets)
+        raise NotImplementedError
+    
+    technologies_parsed = fetch_api(logger)
+
+    """ 
+    TODO
+        * What do I return for production mode that are not even on the island ? 
+            The main README says "The production values should never be negative. Use None, or omit the key if a specific production mode is not known." but doesn't answer this question
+    """
+    return {
+      'countryCode': 'AUS-TAS-KI',
+      'datetime': arrow.now(tz='Australia/Currie').datetime,
+      'production': {
+          'biomass': 0,
```

Comment:
The showcase has an icon that shows if they're using biodiesel (which they currenlty aren't), would there be a way to detect this so it can be switched from oil to biodiesel if they start using that?

Reply:
Mmh, I didn't consider biodiesel as counting as biomass... Is it undoubtedly ?

With no biodiesel currently being used, I'm not sure how to make sense of the payload : 
```
  "biodiesel": {
    "id": "biodiesel",
    "title": "Biodiesel",
    "value": 0,
    "unit": null,
    "percent": 0,
    "status": false
  }
```
My intuition is that when in use, `percent` is the percentage of bio-diesel blended with regular diesel in order to generate the total diesel-generated power, so I would have to subdivide the total diesel energy made into biomass and diesel. Or maybe the `value` and `unit` would be directly in kW. 

I suggest for now to add a TODO block for that. What do you think ?

Reply:
I've looked through the source code, and the only file where biodiesel is mentioned is https://data.ajenti.com.au/KIREIP/files/dashboard.js. The biodiesel code starts at line 170, and the only key from the data that's used is `percent`. If biodiesel and diesel values were reported separately and then added up to the total kW for the diesel gauge, I'd expect biodiesel to be mentioned elsewhere as well (in order to be able to do diesel.value + biodiesel.value).

I've also tried looking for 'diesel' and 'percent', but none of the results suggest anything like _diesel.value + biodiesel.value_ is done.  So I'm 99.99% sure that the total for diesel includes all biodiesel, so the correct thing to do would be
```
oil = diesel * (1- biodiesel.percent/100)
biomass = diesel * biodiesel.percent/100
```

Could you have a look if other things (e.g. wind and solar) have a unit reported when not generating? If so, that would further confirm my suspicion that  `value` and `unit` will always stay null for biodiesel, and are only included in order to have a consistent data structure.

Reply:
> I've also tried looking for 'diesel' and 'percent', but none of the results suggest anything like _diesel.value + biodiesel.value_ is done. So I'm 99.99% sure that the total for diesel includes all biodiesel, so the correct thing to do would be
> 
> ```
> oil = diesel * (1- biodiesel.percent/100)
> biomass = diesel * biodiesel.percent/100
> ```
> 

It makes sense with the presentation of the plant saying that the biodiesel is blended with regular diesel : https://www.hydro.com.au/docs/default-source/clean-energy/hybrid-energy-solutions/king_island.pdf

> Could you have a look if other things (e.g. wind and solar) have a unit reported when not generating? If so, that would further confirm my suspicion that `value` and `unit` will always stay null for biodiesel, and are only included in order to have a consistent data structure.

Yeah I found from one of my test payloads an instance of this : 
```
    {
      "id": "solar",
      "title": "Solar",
      "value": 0,
      "unit": "kW",
      "percent": 0,
      "status": false
    },
    {
      "burn": false,
      "id": "diesel",
      "title": "Diesel",
      "value": 0,
      "unit": "kW",
      "percent": 0,
      "status": false
    },
```

Reply:
See the next commits, especially 219590f